backend/transcribee_backend/routers/document.py

```diff
@@ -191,3 +204,47 @@ async def websocket_endpoint(
         document=document, websocket=websocket, session=session
     )
     await connection.run()
+
+
+@document_router.post("/{document_id}/add_media_file/")
+def add_media_file(
+    document_id: uuid.UUID,
+    tags: list[str] = Form(),
+    file: UploadFile = File(),
+    session: Session = Depends(get_session),
+    authorized_worker: Worker = Depends(get_authorized_worker),
+) -> ApiDocument:
+    statement = select(Task).where(
+        Task.document_id == document_id, Task.task_type == TaskType.REENCODE
+    )
+    task = session.exec(statement).one_or_none()
+    if task is None:
```

Comment:
for what do we need the task here?

Reply:
This is a permission check: the worker is only allowed to upload media if it is assigned to the reencode task